### PR TITLE
Add support for overprovisioning mode

### PR DESCRIPTION
**Pull request by mulkieran**
Related to https://github.com/stratis-storage/stratisd/issues/2814

* Adds ability to set overprovisioning mode
* Display overprovisioning status in the pool listing
* Displays an alert if the pool is out of allocatable space
* Allows to specify that the pool should not overprovision on creation

### CODE REVIEW

**Review by drckeefe — APPROVED** [review]

File: docs/stratis.txt [hunk 1]
```diff
@@ -80,6 +80,11 @@ pool unbind <(clevis|keyring)> <pool name> ::
 pool set-fs-limit <pool name> <amount> ::
      Set the limit on the number of file systems allowed per-pool. This number
      may only be increased from its current value.
+pool overprovisoin <pool name> <(yes|no)> ::
```
Comment (jbaublitz) [inline docs/stratis.txt:83]: There's a typo here.

File: src/stratis_cli/_error_codes.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+
+class PoolMaintenanceErrorCode(IntEnum):
+    """
+    Maintenance error codes for the pool.
+    """
+
+    NO_IPC_REQUESTS = 1
+    NO_POOL_CHANGES = 2
+    READ_ONLY = 3
```
Comment (jbaublitz) [inline src/stratis_cli/_error_codes.py:28]: This appears to be lingering from when we removed it from stratisd

**Review by jbaublitz — CHANGES_REQUESTED** [review]

File: src/stratis_cli/_error_codes.py (hunk 2 above)
Comment (mulkieran) [reply src/stratis_cli/_error_codes.py:28]: I'm addressing that in #869

**Comment by mulkieran** [comment]
rebased, conflicts resolved.

**Comment by mulkieran** [comment]
rebased

**Review by drckeefe — APPROVED** [review]

**Review by jbaublitz — APPROVED** [review]

**Pull request merged**